assets/js/shortcake-bakery.js

```diff
@@ -0,0 +1,48 @@
+(function( $ ) {
+
+	var debounce = function(func, wait, immediate) {
+		var timeout;
+		return function() {
+			var context = this, args = arguments;
+			var later = function() {
+				timeout = null;
+				if (!immediate) {
+					func.apply(context, args);
+				}
+			};
+			var callNow = immediate && !timeout;
+			clearTimeout(timeout);
+			timeout = setTimeout(later, wait);
+			if ( callNow ) {
+				func.apply(context, args);
+			}
+		};
+	};
+
+	var responsiveElements = function() {
+		// Don't use cached elements because YouTube injects new iframes
+		$('.shortcake-bakery-responsive').each(function(){
+			var el = $(this),
+				parentWidth = el.parent().width();
+
+			// Inside an iframe
+			if ( window.self !== window.top ) {
+				parentWidth = parent.innerWidth;
+			}
+
+			var trueHeight = el.data('true-height') ? el.data('true-height') : 360;
+			var trueWidth = el.data('true-width') ? el.data('true-width') : 640;
```

Comment:
> Should this calculation really go here, or should it be up to the shortcodes to listen to?

Then each shortcode has to support it? It's an abstraction that's worked for several shortcodes in our theme.